### PR TITLE
Update our analytics plugin to report paths and query params properly…

### PR DESCRIPTION
**Pull request by mcstover**
…. Add click tracking on lend filter menu and sections. Add loans-shown tracking on lend filter for results shown. The current filter selections are present in urls reported with events. Add pageview tracking when paginating on lend filter.

### CODE REVIEW

File: src/plugins/kv-analytics-plugin.js [hunk 1]
```diff
@@ -10,28 +10,38 @@ export default Vue => {
 			snowplowLoaded = inBrowser && typeof window.snowplow === 'function';
 		},
 		pageview: (to, from) => {
-			const toUrl = window.location.origin + to.path;
-			const asyncFromUrl = window.location.origin + from.path;
+			if (!inBrowser) return false;
+
+			let toUrl = window.location.href;
+			let fromUrl = document.referrer;
```
Comment (mcstover) [inline src/plugins/kv-analytics-plugin.js]: Exit if not in the browser... or set default to + from url properties for use on server rendered pageview events.

File: src/plugins/kv-analytics-plugin.js [hunk 2]
```diff
@@ -10,28 +10,38 @@ export default Vue => {
...
+			// update urls for async page changes
+			if (to.matched && to.matched.length && from.matched && from.matched.length) {
+				toUrl = window.location.origin + to.fullPath;
+				fromUrl = window.location.origin + from.fullPath;
+			}
```
Comment (mcstover) [inline src/plugins/kv-analytics-plugin.js]: Update our to and from urls if we are going between Vue pages that have Vue Router information.

File: src/plugins/kv-analytics-plugin.js [hunk 3]
```diff
@@ -10,28 +10,38 @@ export default Vue => {
...
 
 			// Google Analytics Pageview
 			if (gaLoaded) {
-				if (to.path) {
-					window.ga('set', 'page', to.path);
+				let gaPath = `${window.location.pathname}${window.location.search || ''}`;
+				if (to.matched && to.matched.length) {
+					gaPath = to.fullPath;
 				}
+				window.ga('set', 'page', gaPath);
```
Comment (mcstover) [inline src/plugins/kv-analytics-plugin.js]: GA want's only the path not the full url so we build and apply it here.

File: src/pages/Lend/Filter/FilterComponents/FilterMenuSection.vue [hunk 4]
```diff
@@ -38,6 +38,12 @@ export default {
 	methods: {
 		toggleMenu() {
 			this.open = !this.open;
+
+			const eventAction = `${this.open ? 'open' : 'close'}-${this.title.toLowerCase()}-filter-section`;
+			this.$kvTrackEvent(
+				'Lending',
+				eventAction.replace(/ /g, '-')
+			);
```
Comment (ksekhri) [inline src/pages/Lend/Filter/FilterComponents/FilterMenuSection.vue]: Cool

**Pull request merged**